audplot/core/api.py

```diff
@@ -0,0 +1,178 @@
+import typing
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+import audmetric
+
+
+def confusion_matrix(
+        truth: typing.Union[typing.Sequence, pd.Series],
+        prediction: typing.Union[typing.Sequence, pd.Series],
+        *,
+        labels: typing.Sequence = None,
+        percentage: bool = True,
+):
+    r"""Confusion matrix between ground truth vs. predicted labels.
+
+    The confusion matrx is calculated by :mod:`audmetric.confusion_matrix`.
+
+    Args:
+        truth: truth values
+        prediction: predicted values
+        labels: labels to be included in confusion matrix
+        percentage: if ``True`` present the confusion matrix
+            with percentage values instead of absolute numbers
+
+    Example:
+        .. plot::
+            :context: reset
+            :include-source: false
+
+            from audplot import confusion_matrix
+
+        .. plot::
+            :context: close-figs
+
+            >>> truth = ['A', 'A', 'A', 'B', 'B', 'B', 'C', 'C', 'C']
+            >>> prediction = ['A', 'A', 'B', 'B', 'C', 'C', 'A', 'A', 'C']
+            >>> confusion_matrix(truth, prediction)
+
+    """
+    sns.set()  # get prettier plots
+
+    labels = audmetric.core.utils.infer_labels(truth, prediction, labels)
+
+    cm = audmetric.confusion_matrix(
+        truth,
+        prediction,
+        labels=labels,
+        normalize=percentage,
+    )
+    if percentage:
+        for idx, row in enumerate(cm):
+            if np.sum(row) == 0:
+                cm[idx] = np.NaN
```

Comment:
Results in `IndexError: Inconsistent shape between the condition and the input (got (4, 1) and (4,))`

To fix it we could replace with `cm[idx] = [np.NaN] * len(labels)`, but the confusion matrix we get then looks not nice:

![image](https://user-images.githubusercontent.com/10383417/122525183-45e00080-d019-11eb-8c61-cfa72fa6cd3a.png)

I suggest we replace with:

```python
            if np.sum(row) != 0:
                cm[idx] /= np.sum(row)
```

then we get:

![image](https://user-images.githubusercontent.com/10383417/122525338-6dcf6400-d019-11eb-85b0-f87b7d2bc20e.png)

This also shows that we should target a 100% coverage I think :)

Reply:
What exactly are you doing here, or how do you trigger this?
I did not really understand what `if np.sum(row) != 0` means.

Reply:
Would you add it as another example, or should we add a test function then?

Reply:
> What exactly are you doing here, or how do you trigger this?

I added `labels=['A', 'B', 'C', 'D']` to the docstring example to introduce a class we have no samples for.

Reply:
> Would you add it as another example, or should we add a test function then?

I would add two examples to the doctring, one with `percentage=False` and one with `percentage=True`. If you then apply the fix I suggest above (i.e. not assign `np.NaN` but leave rows with 0s as 0) then we should have a full code coverage.

Reply:
I'm a little bit confused. If I see it corect you are suggesting to replace:

```python
            if np.sum(row) == 0:
                cm[idx] = np.NaN
            else:
                cm[idx] /= np.sum(row)
```

with 
```python
            if np.sum(row) != 0:
                cm[idx] /= np.sum(row)
```
correct?

Reply:
Yes, i.e. if all entries of a row are 0 we simply leave them 0.

Reply:
Oh, but wait a moment... why do we have to divide again at all? If we set `normalize=True` when calling `audmetric.confusion_matrix` we should already get normalized values... now I'm confused :)

Reply:
Actually, it should be:

```python
    if percentage:
        fmt = '.0%'
    else:
        fmt = 'd'
```

Reply:
or: `fmt = '.0%' if percentage else 'd'` then we get full code coverage without a second example :)

Reply:
Yes, seems like we don't need it here.

BTW, in `audmetric` we do the normalization again slightly different: https://github.com/audeering/audmetric/blob/master/audmetric/core/api.py#L173-L177

Reply:
Yes, it could also be simplified to:

```python
    if normalize:
        for idx, row in enumerate(matrix):
            if np.sum(row) != 0:
                row_sum = float(np.sum(row))
                matrix[idx] = [x / row_sum for x in row]
```

Reply:
I removed it and added another plotting example.

I'm trying to make the figure sizes a little bit smaller at the moment.

Reply:
> I'm trying to make the figure sizes a little bit smaller at the moment.

yes, that was another comment I wanted to make :)

Reply:
At the moment I have:

![image](https://user-images.githubusercontent.com/173624/122530189-842bee80-d01e-11eb-8646-a7a2ba920d06.png)

Is this still too big?

I tried to center align the images, but did not succeed.

Reply:
For the confusion matrix it could be smaller, but for other plots this size seems right and it probably looks better if we stick to the same size. I don't mind that it's not centered.

Reply:
OK, then let's stick to a common size for now.

In the usage section, there is an example that shows you how to change the size.